crates/wallet/tests/common.rs

```diff
@@ -208,23 +208,17 @@ pub fn feerate_unchecked(sat_vb: f64) -> FeeRate {
     FeeRate::from_sat_per_kwu(sat_kwu)
 }
 
-/// Simulates confirming a tx with `txid` at the specified `position` by inserting an anchor
-/// at the lowest height in local chain that is greater or equal to `position`'s height,
-/// assuming the confirmation time matches `ConfirmationTime::Confirmed`.
-pub fn insert_anchor_from_conf(
-    wallet: &mut Wallet,
-    txid: Txid,
-    position: ChainPosition<ConfirmationBlockTime>,
-) {
-    if let ChainPosition::Confirmed(anchor) = position {
-        wallet
-            .apply_update(Update {
-                tx_update: tx_graph::TxUpdate {
-                    anchors: [(anchor, txid)].into(),
-                    ..Default::default()
-                },
+/// Simulates confirming a tx with `txid` by applying an update to the wallet containing
+/// the given `anchor`. Note: to be considered confirmed the anchor block must exist in
+/// the current active chain.
+pub fn insert_anchor(wallet: &mut Wallet, txid: Txid, anchor: ConfirmationBlockTime) {
+    wallet
```

Comment:
Can possibly add an assertion here that the anchor height is not greater than the current chain tip.